pkg/oc/bootstrap/docker/up.go

```diff
@@ -324,26 +317,54 @@ func (config *ClientStartConfig) Bind(flags *pflag.FlagSet) {
 	config.CommonStartConfig.Bind(flags)
 }
 
-func (c *CommonStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command) error {
+func (c *CommonStartConfig) Complete(f *osclientcmd.Factory, cmd *cobra.Command, out io.Writer) error {
 	c.originalFactory = f
 	c.command = cmd
 
+	// do some defaulting
 	if len(c.ImageVersion) == 0 {
 		c.ImageVersion = defaultImageVersion()
 	}
 
-	c.addTask(simpleTask("Checking OpenShift client", c.CheckOpenShiftClient))
+	// do some struct initialization next
+
+	// used for some pretty printing
+	taskPrinter := NewTaskPrinter(getDetailedOut(out))
 
-	c.addTask(conditionalTask("Create Docker machine", c.CreateDockerMachine, func() bool { return c.ShouldCreateDockerMachine }))
+	// create a docker machine if we need one
+	if c.ShouldCreateDockerMachine {
+		// this default only gets set if we need to create a dockermachine.
+		if len(c.DockerMachine) == 0 {
+			c.DockerMachine = defaultDockerMachineName
+		}
+
+		taskName := "Create Docker machine"
+		taskPrinter.StartTask(taskName)
+		fmt.Fprintf(out, "Creating docker-machine %s\n", c.DockerMachine)
+		if err := CreateDockerMachine(c.DockerMachine); err != nil {
+			return taskPrinter.ToError(err)
+		}
+		taskPrinter.Success()
+	}
 
 	// Get a Docker client.
 	// If a Docker machine was specified, make sure that the machine is running.
 	// Otherwise, use environment variables.
-	c.addTask(simpleTask("Checking Docker client", c.GetDockerClient))
-
+	{
```

Comment:
why this block?

Reply:
> why this block?

Just containment.  Makes sure nothing leaks.  I suspect cluster up will experience some radical refactoring over the next little while.